aiosmtpd/smtp.py

```diff
@@ -624,6 +631,38 @@ async def smtp_RSET(self, arg):
         status = await self._call_handler_hook('RSET')
         await self.push('250 OK' if status is MISSING else status)
 
+    def _parse_xclient_attributes(self, attributes):
+        """ Parse key-value arguments and return dict on success.
+        Raise ValueError on wrong arguments.
+        """
+        result = {}
+        for kv in attributes.split(' '):
+            try:
+                key, value = kv.split('=')
+            except ValueError as err:
+                raise ValueError('bad command parameter syntax') from err
+            if key not in XCLIENT_ATTRIBUTES:
+                raise ValueError(
+                    '{} is not a valid XCLIENT attribute'.format(key))
+            result[key] = value
+        return result
```

Comment:
Can we reuse [the `_getparams` method](https://github.com/aio-libs/aiosmtpd/blob/16a89fe0054b3c04ccf204631f8fdc9cc85f95de/aiosmtpd/smtp.py#L674)?

Attribute validation can then be performed in the `smtp_XCLIENT` method based on results returned by `_getparams`.